src/ChannelServer/World/Entity.cs

```diff
@@ -10,5 +10,7 @@ public interface IEntity
 		int Hp { get; }
 		Map Map { get; }
 		Position Position { get; }
+		Direction Direction { get; }
+		float Speed { get; }
```

Comment:
My original thought was to use Speed for the base speed of an entity, while GetSpeed, like in Character provides the actual Speed, factoring in buffs and the like. (A separate method to indicate that something is calculated on the fly.) If you change that, you should remove the GetSpeed method entirely.